From: Takashi Iwai <tiwai@suse.de>
Subject: ALSA: emu10k1: Fix kABI breakage
Patch-mainline: Never, SLE15 only
References: bsc#1093027

The recent fixes for IOMMU broke kABI due to the removed / added fields
to snd_emu10k1.  Work around it via usual __GENKSYMS__ trick.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/sound/emu10k1.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/include/sound/emu10k1.h
+++ b/include/sound/emu10k1.h
@@ -1710,7 +1710,6 @@ struct snd_emu10k1 {
 	unsigned int ecard_ctrl;		/* ecard control bits */
 	unsigned int address_mode;		/* address mode */
 	unsigned long dma_mask;			/* PCI DMA mask */
-	bool iommu_workaround;			/* IOMMU workaround needed */
 	unsigned int delay_pcm_irq;		/* in samples */
 	int max_cache_pages;			/* max memory size / PAGE_SIZE */
 	struct snd_dma_buffer silent_page;	/* silent page */
@@ -1719,6 +1718,7 @@ struct snd_emu10k1 {
 	struct snd_dma_buffer p16v_buffer;
 
 	struct snd_util_memhdr *memhdr;		/* page allocation list */
+	struct snd_emu10k1_memblk *reserved_page;	/* reserved page */
 
 	struct list_head mapped_link_head;
 	struct list_head mapped_order_link_head;
@@ -1803,6 +1803,9 @@ struct snd_emu10k1 {
 	bool suspend;
 #endif
 
+#ifndef __GENKSYMS__
+	bool iommu_workaround;			/* IOMMU workaround needed */
+#endif
 };
 
 int snd_emu10k1_create(struct snd_card *card,
